static/js/subs.js

```diff
@@ -259,7 +259,7 @@ exports.update_settings_for_subscribed = function (sub) {
         exports.add_sub_to_table(sub);
     }
 
-    if (stream_edit.is_sub_settings_active(sub)) {
+    if (stream_edit.is_sub_settings_active(sub) && sub.invite_only) {
```

Comment:
We don't want rerender subscribers, cause that will rerender subscribers in alphabetical order. But in case of private stream we want to, cause when user subscribe to private stream, there is no existing subscriber list so we need to rerender whole subscriber list. So if user is subscribed to private stream(when some other private stream's subscribed user add them) we may not have currently subscribed user on top of list, if user has stream subscription modal opened.

Reply:
This doesn't seem like the right fix.  I'm merging it for now since it's better than the status quo, but if we think the correct behavior is to prepend the user, we should make that work, private stream or no (we could do that easily by extending the state).  The other option would be to always sort the current user at the top, which might be a better solution...

Can you open an issue for this question and mention Rishi to get an opinion on whether we should just always sort the current user to the top?

Reply:
Sure, opened #9027